### ISSUE TITLE
Caching not implemented for OpenAI LLM stream usage

### ISSUE THREAD

**Issue by Shakahs** [langchain-ai/langchainjs#3582]
Caching is implemented for OpenAI LLM `invoke` codepath but not `stream`.
Caching should be implemented for both usage styles. 

Minimal reproduction:
Passes: `should update cache when run via invoke` 
Fails: `should update cache when run via stream` 
 
```typescript
import { jest, describe, beforeEach, it, expect, beforeAll } from '@jest/globals';
import { ChatOpenAI } from 'langchain/chat_models/openai';
import { InMemoryCache } from '@langchain/core/caches';
import { OpenAI } from 'langchain/llms/openai';


describe('OpenAI LLM', () => {
    let llm;
    let cacheStorage;

    beforeEach(() => {
        cacheStorage = new Map();
        llm = new OpenAI({
            modelName: 'gpt-4',
            openAIApiKey: 'sk-xxxxxxxxx',
            cache: new InMemoryCache(cacheStorage),
        });
        jest.spyOn(llm.cache, 'lookup');
        jest.spyOn(llm.cache, 'update');
    });

    it('should update cache when run via invoke', async () => {
        const result = await llm.invoke('How are you today');
        expect(cacheStorage.size).toEqual(1);
        expect(llm.cache.update).toHaveBeenCalled();
    });

    it('should update cache when run via stream', async () => {
        const result = await llm.stream('How are you today');
        for await (const i of result) {
            console.log(i);
        }
        expect(cacheStorage.size).toBeGreaterThanOrEqual(1);
        expect(llm.cache.update).toHaveBeenCalled();
    });
});

```

**Comment by jacoblee93** [comment]
What would be the expected behavior for this? That it immediately streams back a single cached chunk?

**Comment by RoyalSix** [comment]
This is still an issue, would be nice to be able to cache streamed results

**Comment by jacoblee93** [comment]
Agreed, thanks for continuing to push on this.

What would the ideal behavior be? Everything yielded in one large chunk?

**Comment by RoyalSix** [comment]
Hi Jacob, I guess if it was a stream it would have to be somewhat artificially created by arbitrarily spacing out the chunks, so I would say everything yielded in one large chunk makes more sense

**Comment by rasmushjulskov** [comment]
One large chunk would be fine 👍

**Comment by EliiseS** [comment]
I'd also love to have caching with streaming

**Comment by racinger** [comment]
same here

**Comment by jasoncsullivan** [comment]
@jacoblee93 could this be accomplished with EventStream? Retain the streaming behavior by pushing on_llm_stream events to the client side, while only caching the result of on_llm_end?

**Comment by jacoblee93** [comment]
Not sure what you mean? Basically cache and then emit all the individual chunks?

**Comment by EliiseS** [comment]
Does this work with the python SDK? Because I also notice it's missing there as well. How are the python and JS SDKs linked?

**Comment by Shakahs** [comment]
> Basically cache and then emit all the individual chunks?

Yes. This would be the simplest, most intuitive implementation. A caching layer shouldn't be transforming data, transforming the data introduces additional complexity, and I don't a downside to just storing and emitting the original chunks. 

It also breaks the API contract.

**Comment by necjamahim** [comment]
> Not sure what you mean? Basically cache and then emit all the individual chunks?
I think caching the response and emitting individual chunks would be good. This way we save LLM calls and output the response downstream compatible. 

Thats how Im gonna implement it for my use case. Otherwise I would have to run invoke to lookup the cache and then simulate streaming in frontend....

**Comment by kunalhexus** [comment]
I also need this

**Comment by saitaiky** [comment]
Just wondering if there’s been any update on this feature. Currently, when a response is served from the cache, it doesn’t get streamed back. My expectation would be that the `messages` stream event still gets emitted, with the cached value returned as the response.

When building a conversational application, receiving only the `updates ` stream makes it difficult to distinguish between a status update and an actual message response (even from cache). This ambiguity complicates the logic for handling streamed interactions.

It would be great if the stream behaviour could be consistent regardless of whether the response is cached or freshly generated.
